cmd/ttn-lw-cli/commands/applications_packages.go

```diff
@@ -0,0 +1,240 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"go.thethings.network/lorawan-stack/cmd/ttn-lw-cli/internal/api"
+	"go.thethings.network/lorawan-stack/cmd/ttn-lw-cli/internal/io"
+	"go.thethings.network/lorawan-stack/cmd/ttn-lw-cli/internal/util"
+	"go.thethings.network/lorawan-stack/pkg/ttnpb"
+)
+
+var (
+	selectApplicationPackageAssociationsFlags = util.FieldMaskFlags(&ttnpb.ApplicationPackageAssociation{})
+	setApplicationPackageAssociationsFlags    = util.FieldFlags(&ttnpb.ApplicationPackageAssociation{})
+)
+
+func applicationPackageAssociationIDFlags() *pflag.FlagSet {
+	flagSet := &pflag.FlagSet{}
+	flagSet.String("application-id", "", "")
+	flagSet.String("device-id", "", "")
+	flagSet.String("f-port", "", "")
+	return flagSet
+}
+
+func getApplicationPackageAssociationID(flagSet *pflag.FlagSet, args []string) (*ttnpb.ApplicationPackageAssociationIdentifiers, error) {
+	applicationID, _ := flagSet.GetString("application-id")
+	deviceID, _ := flagSet.GetString("device-id")
+	fport, _ := flagSet.GetUint32("f-port")
```

Comment:
Why not `Uint8`?

We use `uint32` in protos because it doesn't support `uint8`, but that doesn't mean we should use them in the application layer